src/lib/user-features.ts

```diff
@@ -142,25 +146,35 @@ const deleteOldData = (): void => {
 	removeCookie({ name: ONE_OFF_CONTRIBUTION_DATE_COOKIE });
 };
 
-const requestNewData = () =>
-	fetchJson(
-		`${
-			window.guardian.config.page.userAttributesApiUrl ??
-			'/USER_ATTRIBUTE_API_NOT_FOUND'
-		}/me`,
-		{
-			mode: 'cors',
-			credentials: 'include',
-		},
-	)
-		.then((response) => {
-			if (!validateResponse(response)) {
-				throw new Error('invalid response');
-			}
-			return response;
-		})
-		.then(persistResponse)
-		.catch(noop);
+const requestNewData = () => {
+	return getAuthStatus()
+		.then((authStatus) =>
+			authStatus.kind === 'SignedInWithCookies' ||
+			authStatus.kind === 'SignedInWithOkta'
+				? authStatus
+				: Promise.reject('The user is not signed in'),
+		)
+		.then((signedInAuthStatus) => {
+			return fetchJson(
+				`${
+					window.guardian.config.page.userAttributesApiUrl ??
+					'/USER_ATTRIBUTE_API_NOT_FOUND'
+				}/me`,
+				{
+					mode: 'cors',
+					...getOptionsHeadersWithOkta(signedInAuthStatus),
+				},
+			)
+				.then((response) => {
+					if (!validateResponse(response)) {
+						throw new Error('invalid response');
+					}
+					return response;
+				})
+				.then(persistResponse)
+				.catch(noop);
+		});
```

Comment:
```suggestion
		.then((signedInAuthStatus) => fetchJson(
		        `${
			        window.guardian.config.page.userAttributesApiUrl ??
			        '/USER_ATTRIBUTE_API_NOT_FOUND'
		        }/me`,
		        {
			        mode: 'cors',
			        ...getOptionsHeadersWithOkta(signedInAuthStatus),
		        },
	        ))
		.then((response) => {
			if (!validateResponse(response)) {
				throw new Error('invalid response');
			}
			return response;
		})
		.then(persistResponse)
		.catch(noop);
```

You could make use of [promise chaining](https://blog.greenroots.info/javascript-promise-chain-the-art-of-handling-promises#heading-promise-chain-rule-2) here if you wanted to reduce verbosity here!

Reply:
This is a line for line copy of the same function from frontend - so I'd probably prefer to keep the two aligned with one another!